src/Spice86/Emulator/CPU/RegistersHolder.cs

```diff
@@ -41,10 +41,14 @@ public class RegistersHolder {
         return $"{reg16[..1]}{suffix}";
     }
 
-    public ushort GetRegister(int index) {
+    public uint GetRegister32(int index) {
         return _registers[index];
     }
 
+    public ushort GetRegister(int index) {
+        return (ushort)(GetRegister32(index) & 0xFFFF);
+    }
+
     public byte GetRegister8H(int regIndex) {
         return ConvertUtils.ReadMsb(GetRegister(regIndex));
```

Comment:
How about creating methods like ConvertUtils.ReadMsw and ConvertUtils.WriteMsw ? The could be used in other parts of the code in the future, you never know :)